lib/dms.js

```diff
@@ -46,9 +46,47 @@ class DmsModel {
     await Promise.all(datapackage.resources.map(async resource => {
       resource.views = await this.getResourceViews(resource.id)
     }))
+
+    // Fetch dictionary fields for the resources
+    await Promise.all(datapackage.resources.map(async resource => {
+      //TODO Check resource type and make
+      resource.schema = await this.getResourceSchema(resource.id)
+    }))
     return datapackage
   }
 
+  async getResourceSchema(resourceId) {
+    try{
+      const action = 'datastore_search'
+      let url = new URL(resolve(this.api, action))
+      let response = await fetch(url, {
+        method: 'POST',
+        headers: { 
+          'Content-Type': 'application/x-www-form-urlencoded'
+        },
+        body: JSON.stringify({
+          'resource_id' : resourceId
```

Comment:
We need to add another attribute here to decrease response size. By default, it returns 100 rows from the dataset, however, we don't need it. So you can pass `limit: 0` option here.